File: DCP-03-25/Minimum-Recolors-to-Get-K-Consecutive-Black-Blocks.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int r=k;
        int c=count(blocks.begin(),blocks.begin()+k,'B');
        int res=k-c;
        while(r<blocks.size()){
            c-=blocks[r-k]=='B';
            c+=blocks[r++]=='B';
            res=min(res,k-c);
        }
    return res;
    }
};

File: DCP-03-25/Minimum-Recolors-to-Get-K-Consecutive-Black-Blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        r=k
        c=0
        for i in range(k):
            c+=blocks[i] =='B'

        res=k-c

        while r <len(blocks) :
            
            c-= 1 if blocks[r-k]=='B' else  0
            c+= 1 if blocks[r]=='B' else  0
       
            res=min(res,k-c)
            r+=1
            
        return res

            